nuxeo-platform-3d-convert/src/main/java/org/nuxeo/ecm/platform/threed/convert/BaseBlenderConverter.java

```diff
@@ -93,7 +127,14 @@ protected boolean isThreeDFile(File file) {
             }
 
         }
-        return expandedFiles;
+        List<String> fileNames = new ArrayList<>();
+        // sort files by name alphabetically
+        files3d.sort(Comparator.comparing(File::getName));
+        filesOther.sort(Comparator.comparing(File::getName));
+        // return all the file names, 3D files alphabetically sorted, followed by non 3D files alphabetically sorted
+        fileNames.addAll(files3d.stream().map(File::getAbsolutePath).collect(Collectors.toList()));
+        fileNames.addAll(filesOther.stream().map(File::getAbsolutePath).collect(Collectors.toList()));
```

Comment:
You could do the sort on each stream.
Furthermore, I think it's possible to concatenate the two streams in order to collect the list only once (and keep your behavior).